### PR TITLE
Add protoc-gen-jsonpb

### PR DESCRIPTION
**Pull request by ajm188**
### Why?

Protos. They're great!! But sometimes you have an application that doesn't speak grpc everywhere, and sometimes you have a struct that has a protobuf enum in it. When you go to `json.Marshal` that however  ... you end up with something like:

```
{
  "hostname": "tablet-whatever",
  "tablet_type": 2,
}
```

instead of the more readable (and less requiring-your-client-to-hardcode-int-to-string-maps—y)
```
{
  "hostname": "tablet-whatever",
  "tablet_type": "REPLICA",
}
```

This results in sadness.

What you would _like_ to do is just use [github.com/golang/protobuf/jsonpb](https://pkg.go.dev/github.com/golang/protobuf@v1.4.2/jsonpb) to spin up a marshaler and call `m.MarshalPB(&mything)`, but that only works if `mything` is itself a protoc-generated struct/type/etc. If you want to embed a field of that type and marshal the outer struct, you'll end up with something like:

```
cannot use mything (variable of type *MyType) as protoiface.MessageV1 value in argument to m.Marshal: missing method ProtoMessage
```

This results in even more sadness.

One way to fix this would be to create a second file alongside your generated pb.go file where you declare, _for every protobuf struct in your generated code_:

```go
func (m *MyType) MarshalJSON() ([]byte, error) {
    json := jsonpb.Marshaler{}
    b := bytes.Buffer{}

    if err := json.Marshal(b, m); err != nil {
        return nil, err
    }

    return b.Bytes(), nil
}
```

This allows you to use the stdlib `encoding/json`'s `Marshal()`, and under the hood your type will be marshalled using `jsonpb`.

But this setup is kind of a bummer. The whole point of protoc-gen-go is to *not* have to write boilerplate code for your protobuf types.

What if there was a tool that could modify your AST to add these functions for you? Well, I have good news friend.

### CODE REVIEW

File: jsonpb.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package jsonpb
+
+import (
+	"bytes"
+
+	"github.com/golang/protobuf/jsonpb"
```
Comment (falun) [inline jsonpb.go]: Take it or leave it: For clarity I like to alias packages that come in and share a name with the current package or any global level declarations within the current package.

**Review by falun — COMMENTED** [review]
It feels too presumptuous to Approve this but it looks good :grinning:. Good call on reusing the post-apply to side step needing to process the proto itself for a template approach.

Generally I find the "constructed file" more readable than "writing code by struct" so I'm glad it was straight forward.

**Comment by ajm188** [comment]
Time to make this the ~~main~~ default branch, because I forgot to do that when I set up the repo 😅

**Comment by ajm188** [comment]
"merged" or whatever

**Pull request merged**